tests/models/test_cli.py

```diff
@@ -535,3 +539,54 @@ def test_env_manager_unsupported_value():
             ["--model-uri", "model", "--env-manager=abc"],
             catch_exceptions=False,
         )
+
+
+def test_change_conda_env_root_location(tmp_path, sk_model):
+    env_root1_path = tmp_path / "root1"
+    env_root1_path.mkdir()
+
+    env_root2_path = tmp_path / "root2"
+    env_root2_path.mkdir()
+
+    model1_path = tmp_path / "model1"
+    mlflow.sklearn.save_model(sk_model, str(model1_path), pip_requirements=["scikit-learn==1.0.1"])
+
+    model2_path = tmp_path / "model2"
+    mlflow.sklearn.save_model(sk_model, str(model2_path), pip_requirements=["scikit-learn==1.0.2"])
+
+    for env_root_path, model_path, sklearn_ver in [
+        (env_root1_path, model1_path, "1.0.1"),
+        (
+            env_root2_path,
+            model1_path,
+            "1.0.1",
+        ),  # test the same env created in different env root path.
+        (
+            env_root1_path,
+            model2_path,
+            "1.0.2",
+        ),  # test different env created in the same env root path.
```

Comment:
Curious what should happen after running these? 3 conda environments should be created?

Reply:
If so, does the current test code guarantee that?

Reply:
added test